packages/next-app/src/app/metadata/(components)/MetadataSidebar.tsx

```diff
@@ -0,0 +1,74 @@
+import { Pool } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
+import Link from "next/link";
+import { useNetwork } from "wagmi";
+
+import { Badge } from "#/components/Badge";
+import Sidebar from "#/components/Sidebar";
+import { usePoolMetadata } from "#/contexts/PoolMetadataContext";
+import { networkFor } from "#/lib/gql";
+import { truncateAddress } from "#/utils/truncateAddress";
+
+export function MetadataSidebar() {
+  const { chain } = useNetwork();
+  const { selectedPool, handleSetPool, poolsData } = usePoolMetadata();
+  const network = networkFor(chain!.id.toString());
+
+  return (
+    <Sidebar isFloating>
+      <Sidebar.Header name="Owned pools" />
+      <Sidebar.Content>
+        {poolsData &&
+          poolsData.map((item) => (
+            <Link
+              key={item.id}
+              href={`/metadata/${network}/pool/${item.id}`}
+              onClick={() => handleSetPool(item.id)}
+            >
+              <Sidebar.Item isSelected={item.id === selectedPool}>
+                <OwnedPool
+                  isSelected={item.id === selectedPool}
+                  pool={item as Pool}
+                />
+              </Sidebar.Item>
+            </Link>
+          ))}
```

Comment:
Could we add a loading state here?

Reply:
We currently have a loading state from the main fetch that gathers the pool data. We could pass here using context and using the loading state inside the `Sidebar` but right now I don't think it would be useful as the page is only rendered if the `poolsData` length is bigger than 0. With the current way we are doing the fetching we only update `poolsData` state when the promise is over. So there is no way the Sidebar will be rendered without the complete data

<img width="414" alt="image" src="https://user-images.githubusercontent.com/63298341/230210570-f67b4e15-b518-4d62-98b1-42f1dc6a5e1e.png">

I agree we should have a loading state here, but for that I think we should first check if it's possible to gql to kind of parse the response into chunks and update those chunks into the `poolsData` state.

Does that make sense to you?